packages/config/src/genesisConfig/index.ts

```diff
@@ -22,8 +22,19 @@ export function createICachedGenesis(chainForkConfig: IChainForkConfig, genesisV
   }
 
   return {
-    getDomain(domainType: DomainType, slot: Slot): Uint8Array {
-      const forkInfo = chainForkConfig.getForkInfo(slot);
+    getDomain(fork: phase0.Fork | undefined, domainType: DomainType, slot: Slot): Uint8Array {
```

Comment:
It has to be an optional parameter why not putting it last? 
```suggestion
    getDomain(domainType: DomainType, slot: Slot, stateFork?: phase0.Fork): Uint8Array {
```